be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -71,35 +71,36 @@ Status LakePrimaryIndex::_do_lake_load(Tablet* tablet, const TabletMetadata& met
-                    std::string path = strings::Substitute(
-                            "$0/$1/",
-                            StorageEngine::instance()->get_persistent_index_store()->get_persistent_index_path(),
-                            tablet->id());
-
-                    RETURN_IF_ERROR(
-                            StorageEngine::instance()->get_persistent_index_store()->create_dir_if_path_not_exists(
-                                    path));
-                    _persistent_index = std::make_unique<LakeLocalPersistentIndex>(path);
-                    return ((LakeLocalPersistentIndex*)_persistent_index.get())
-                            ->load_from_lake_tablet(tablet, metadata, base_version, builder);
-                }
-                default:
-                    LOG(WARNING) << "only support LOCAL lake_persistend_index_type for now";
-                    return Status::InternalError("only support LOCAL lake_persistend_index_type for now");
+        auto persistent_index_type = tablet->get_persistent_index_type(base_version);
+        if (persistent_index_type.ok()) {
+            switch (persistent_index_type.value()) {
+            case PersistentIndexTypePB::LOCAL: {
+                // Even if `enable_persistent_index` is enabled,
+                // it may not take effect if is as compute node without any storage path.
+                if (!(StorageEngine::instance()->check_valid_storage_paths())) {
+                    LOG(WARNING) << "lake_persistent_index_type of LOCAL will not take effect when as cn without any "
+                                    "storage path";
+                    return Status::InternalError(
+                            "lake_persistent_index_type of LOCAL will not take effect when as cn without any storage "
+                            "path");
                 }
-            }
+                std::string path = strings::Substitute("$0/$1/",
+                                                       StorageEngine::instance()
+                                                               ->get_persistent_index_store(tablet->id())
+                                                               ->get_persistent_index_path(),
+                                                       tablet->id());
 
-        } else {
-            LOG(WARNING) << "lake tablet persistent_index will not take effect, for storage_root_path is not set";
-            return Status::InternalError(
-                    "lake tablet persistent_index will not take effect, for storage_root_path is not set");
+                RETURN_IF_ERROR(StorageEngine::instance()
+                                        ->get_persistent_index_store(tablet->id())
+                                        ->create_dir_if_path_not_exists(path));
+                _persistent_index = std::make_unique<LakeLocalPersistentIndex>(path);
+                return ((LakeLocalPersistentIndex*)_persistent_index.get())
```

Comment:
_persistent_index-> load_from_lake_tablet(...)

Reply:
The type of `_persistent_index` is `std::unique_ptr<PersistentIndex> ` can not call  `load_from_lake_tablet` directly

Reply:
Is it better to use type casting?